python/dgl/graphbolt/utils/sample_utils.py

```diff
@@ -164,6 +164,7 @@ def unique_and_compact_node_pairs(
         torch.Tensor,
         Dict[str, torch.Tensor],
     ] = None,
+    deduplicate=True,
```

Comment:
We should not add deduplicate parameter here, the name of the method unique_and_compact_node_pairs means it will be deduplicated. We will need a new method here.